plan/src/test/java/org/apache/flink/agents/plan/serializer/ActionJsonSerializerTest.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.agents.plan.serializer;
+
+import org.apache.flink.agents.api.Event;
+import org.apache.flink.agents.api.InputEvent;
+import org.apache.flink.agents.api.OutputEvent;
+import org.apache.flink.agents.plan.Action;
+import org.apache.flink.agents.plan.JavaFunction;
+import org.apache.flink.agents.plan.PythonFunction;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** Test for {@link ActionJsonSerializer}. */
+public class ActionJsonSerializerTest {
+    @Test
+    public void testSerializeJavaFunction() throws Exception {
+        // Create a JavaFunction
+        JavaFunction function =
+                new JavaFunction(
+                        "org.apache.flink.agents.plan.TestAction",
+                        "legal",
+                        new Class[] {InputEvent.class});
+
+        // Create an Action
+        Action action = new Action("testAction", function, List.of(InputEvent.class));
+
+        // Serialize the action to JSON
+        String json = new ObjectMapper().writeValueAsString(action);
+
+        // Verify the JSON contains the expected fields
+        assertTrue(json.contains("\"name\":\"testAction\""), "JSON should contain the action name");
```

Comment:
Curious why not do `assertEquals` with whole serialized json? Will that be more readable?